### PR TITLE
[#8786] Move twisted.mail.smtp interfaces into a common file

### PR DESCRIPTION
**Pull request by hawkowl**
https://twistedmatrix.com/trac/ticket/8786

### CODE REVIEW

File: src/twisted/mail/interfaces.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+        @type recipients: C{list} of L{User}
+        @param recipients: A list of the addresses for which this message
+        is bound.
+
+        @rtype: C{str}
+        @return: The full \"Received\" header string.
+        """
+
+    def validateTo(user):
```
Comment (rodrigc) [inline src/twisted/mail/interfaces.py:52]: You should take a pass through the doc string,
and change C{str} -> L{str},   C{list} -> L{list}
C{Deferred} -> L{Deferred}

etc.

File: src/twisted/mail/smtp.py [hunk 2]
```diff
@@ -33,6 +36,29 @@
 except ImportError:
```
Comment (rodrigc) [inline src/twisted/mail/smtp.py:36]: Can you get rid of this?

File: src/twisted/mail/smtp.py [hunk 3]
```diff
@@ -33,6 +36,29 @@
...
     from StringIO import StringIO
```
Comment (rodrigc) [inline src/twisted/mail/smtp.py:37]: Can you get rid of this?

**Comment by rodrigc** [comment]
Address my comments, then merge.

**Comment by ghost** [comment]
@rodrigc @hawkowl 

<hijack>
I see that you guys are working with twisted.mail. I'd like to get fixes for these two issues into a release.

https://twistedmatrix.com/trac/ticket/8285
https://twistedmatrix.com/trac/ticket/8804

**Comment by hawkowl** [comment]
@burrows-labs if you have a PR I'd be happy to look at it, otherwise I guess we'll get to it when we get to it, sadly!

**Comment by rodrigc** [comment]
@burrows-labs the procedure for submitting patches to Twisted changed since you filed 8285 and 8804.  Please follow the new procedure documented here: http://twistedmatrix.com/trac/wiki/TwistedDevelopment#SubmittingaPatch

and reference the pull requests in those tickets.

**Comment by ghost** [comment]
Ok, I will work on some PRs.

Thanks

**Pull request merged**